xbmc/music/MusicDatabase.cpp

```diff
@@ -3615,9 +3615,11 @@ bool CMusicDatabase::GetRolesNav(const std::string& strBaseDir, CFileItemList& i
       CFileItemPtr pItem(new CFileItem(labelValue));
       pItem->GetMusicInfoTag()->SetTitle(labelValue);
       pItem->GetMusicInfoTag()->SetDatabaseId(m_pDS->fv("role.idRole").get_asInt(), "role");
-
-      std::string artistrolepath = StringUtils::Format("musicdb://artists/?roleid=%i", m_pDS->fv("role.idRole").get_asInt());
-      pItem->SetPath(artistrolepath);
+      CMusicDbUrl itemUrl;
```

Comment:
In this case no.  The path to the roles node is `musicdb://roles/` that is what will be in strBaseDir.  But the required role ID, when navigating from a role, is passed as an option value rather than extend the existing 4 parameter (idGenre, IdArtist, idAlbum, IdSong) system. That way both genre and role can be used together.

The items of the roles node are given a path of musicdb://artists/?roleid=n plus any other options from strBaseDir